### PR TITLE
Fix wrong narration in browse mode

### PR DESCRIPTION
**Pull request by krkshitij**
## Current Behavior

In browse mode, screen reader announces same information for all focusable SVG shapes

## New Behavior

In browse mode, screen reader announces respective information for the SVG shapes

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d1228a137bb5f28014bad869d88584ae95ba46da'>d1228a137bb5f28014bad869d88584ae95ba46da</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-charting-91629625-fe59-4a0c-8181-32a38434b3b3.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
[the remark this answers is not in the document]
Comment (krkshitij) [reply change/@fluentui-react-charting-91629625-fe59-4a0c-8181-32a38434b3b3.json]: Created a separate PR for that

File: packages/react-charting/src/components/DonutChart/Arc/Arc.tsx [hunk 2]
```diff
@@ -58,8 +58,8 @@ export class Arc extends React.Component<IArcProps, IArcState> {
           onBlur={this._onBlur}
           opacity={opacity}
           onClick={href ? this._redirectToUrl.bind(this, href) : this.props.data?.data.onClick}
-          aria-labelledby={this.props.calloutId}
-          role="text"
+          aria-label={this._getAriaLabel()}
+          role="img"
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/DonutChart/Arc/Arc.tsx:62]: Is there any downside of using img as role

File: packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx [hunk 3]
```diff
@@ -522,4 +522,11 @@ export class GroupedVerticalBarChartBase extends React.Component<
   private _noLegendHighlighted = () => {
     return this.state.selectedLegend === '' && this.state.activeLegend === '';
   };
+
+  private _getAriaLabel = (point: IGVBarChartSeriesPoint, xAxisPoint: string): string => {
+    const xValue = point.xAxisCalloutData || xAxisPoint;
+    const legend = point.legend;
+    const yValue = point.yAxisCalloutData || point.data;
+    return point.callOutAccessibilityData?.ariaLabel || `${xValue}. ${legend}, ${yValue}.`;
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx:530]: there is a property called stackCalloutAccessibilityData. Are you not using that.

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx [hunk 4]
```diff
@@ -461,4 +460,10 @@ export class MultiStackedBarChartBase extends React.Component<IMultiStackedBarCh
   private _noLegendHighlighted = () => {
     return this.state.selectedLegend === '' && this.state.activeLegend === '';
   };
+
+  private _getAriaLabel = (point: IChartDataPoint): string => {
+    const legend = point.xAxisCalloutData || point.legend;
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx:465]: No x value here?

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 5]
```diff
@@ -706,4 +704,19 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
...
+    const legend = point.legend;
+    const yValue = point.yAxisCalloutData || point.y;
+    const lineLegend = this.props.lineLegendText || 'Line';
+    const lineYValue = point.lineData?.yAxisCalloutData || point.lineData?.y;
+    return (
+      point.callOutAccessibilityData?.ariaLabel ||
+      `${xValue}. ` +
+        (legend ? `${legend}, ` : '') +
+        `${yValue}.` +
+        (typeof lineYValue !== 'undefined' ? ` ${lineLegend}, ${lineYValue}.` : '')
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:719]: what does this line mean

File: packages/react-charting/src/components/VerticalStackedBarChart/VerticalStackedBarChart.base.tsx [hunk 6]
```diff
@@ -724,7 +724,7 @@ export class VerticalStackedBarChartBase extends React.Component<
         });
         const rectFocusProps = !shouldFocusWholeStack && {
           'data-is-focusable': !this.props.hideTooltip,
-          'aria-labelledby': `toolTip${this._calloutId}`,
+          'aria-label': this._getAriaLabel(singleChartData, point),
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/VerticalStackedBarChart/VerticalStackedBarChart.base.tsx:727]: What is the issue with callout content. Is that not always upto date

File: packages/react-charting/src/components/DonutChart/Arc/Arc.tsx (hunk 2 above)
Comment (krkshitij) [reply packages/react-charting/src/components/DonutChart/Arc/Arc.tsx:62]: I couldn't find any. The SVG shapes are better identified by the img role and we have already used it in some of the charts.
Also, 
image

File: packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx (hunk 3 above)
Comment (krkshitij) [reply packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx:530]: Unlike VerticalStackedBarChart, here the focus moves only on individual bars and not on stacks (groups of bars)

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx (hunk 4 above)
Comment (krkshitij) [reply packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx:465]: Charts without axes don't have x value

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 5 above)
Comment (krkshitij) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:719]: Line data is optional. If line data is provided (i.e. not undefined), then include it in aria-label

File: packages/react-charting/src/components/VerticalStackedBarChart/VerticalStackedBarChart.base.tsx (hunk 6 above)
Comment (krkshitij) [reply packages/react-charting/src/components/VerticalStackedBarChart/VerticalStackedBarChart.base.tsx:727]: In browse mode, when the screen reader's cursor is moved to bars using Ctrl + arrow keys, it doesn't trigger focus event and the callout stays hidden with outdated content

image

**Pull request merged**